CodeForPittsburgh/law-reading-robot-data
# Create first version of summarize etl script

## What I did

Created a script for pulling text from all non-summarized revisions in Supabase, summarizing them, and uploading the summaries back to supabase, and integrated that with the existing Summarization code.

This PR contains everything but the actual summarization code -- thus, it primarily deals with interfacing with the database and performing all logic surrounding the summarization itself.

It also contains a yaml file for running the workflow as a Github Action, as well as test data to run and confirm functionality.

## Issue #s Addressed

https://trello.com/c/xcJjkXVG/58-summarization-script-database-interface-component
https://trello.com/c/gImZnLvp/43-create-draft-yaml-file-for-summarization-workflow

## How to test it

1. Check out the branch on your local machine
2. Set up Supabase according to instructions in README `supabase` directory.
3. Insert data from csvs according to instructions in README in `tests/supabase_test_data` directory
4. Insert manual revision text data according to README in `tests/supabase_test_data/test_summarize` directory
5. Add to .env file (created during instructions in step 2) your OpenAI API key, labeled as `OPENAI_API_KEY`
6. Ensure the account connected to the API key has enough credits to run summaries (should be less than a dollar). 
7. Note: if you have to upgrade your OpenAI API account to a paid tier during this process, you will need to generate a new API key and update your .env file with that
8. Run `summarize_etl.py` in `law_reader` directory.
9. Examine results in local dev environment Supabase API Studio
